internal/provider/resource_tfe_workspace_run_task.go

```diff
@@ -297,3 +320,84 @@ func (r *resourceWorkspaceRunTask) UpgradeState(ctx context.Context) map[int64]r
 		},
 	}
 }
+
+func (r *resourceWorkspaceRunTask) supportsStagesProperty() bool {
+	// The Stages property is available in HCP Terraform and Terraform Enterprise v202404-1 onwards.
+	//
+	// The version comparison here can use plain string comparisons due to the nature of the naming scheme. If
+	// TFE every changes its scheme, the comparison will be problematic.
+	return r.capabilities.IsCloud() || r.capabilities.RemoteTFEVersion() > "v202404"
```

Comment:
Probably worth placing the static version into a global SupportedStagesPropertyVersion variable

Reply:
will do

Reply:
Done